### DISCUSSION TITLE
Bug reactive programmation with multiple traces at the same time

### DISCUSSION THREAD

**Discussion by Flooze-IT** [open-telemetry/opentelemetry-java-instrumentation#7320] [Q&A]
Hi everyone,

I have noticed for several days a particular behavior about traces.

For a lit bit of context, i develop my application with a reactive mod by using the webflux library/

Problem:
When i run my application i have some beans which is initialized and they do some reactive process with tracing 

Trace1: Step1 (async) (spanBeans1) -> Step2 (async) (spanBeans2) -> Step3 (async) (spanBeans3)

During this process, i make a request to my service with async steps:

Trace2: Step1 (async) (span1) -> Step2 (async) (span2) -> Step3 (async) (span3)

Result 

I have spans from trace 2 which end up in trace 1 because the two tracks were played in parallel (Reactive programming, multi-threading)

Trace1: spanBeans1 -> span2-> spanBeans2 -> spanBeans3
Trace2: span1 -> span3

Questions: 

I found a solution by passing the span parent in each function but i don't know if this is the right way

The multi threading with reactive programmation and mulitple context is support by open telemetry ?

**Comment by Flooze-IT** [comment]
I notice that when i want to trace inside flatMap and keep tracking the trace, the context is lost and we start a new trace.

**Comment by laurit** [comment]
Create a github issue, include minimal application that reproduces the problem that you observe.
